lib/evmone/eof.cpp

```diff
@@ -771,8 +784,7 @@ std::variant<EOF1Header, EOFValidationError> validate_header(
         container_offsets.emplace_back(static_cast<uint16_t>(offset));
         offset += container_size;
     }
-    // NOTE: assertion always satisfied only as long as initcode limits apply (48K).
-    assert(offset <= std::numeric_limits<uint16_t>::max());
```

Comment:
I think we should keep the assert.

Reply:
This looks still removed?

Reply:
I mean I would still keep it even if it's checked in `validate_section_headers()`

Reply:
ah it moved up a bit and changed form, but the outcome is still the same (I think, please double check if in doubt) - the cast is safe.

Reply:
`assert(container.size() <= MAX_INITCODE_SIZE);` checks that container size is checked outside `validate_header()` and `validate_section_headers()` check that declared containers sizes sum doesn't exceed container size, so the cast is safe, but I would still keep the assertion here (to prevent future errors in case `validate_section_headers()` is changed).

Also currently it's incosistent, you removed this assertion, but kept another one at line 776.

Reply:
> Also currently it's incosistent, you removed this assertion, but kept another one at line 776.

ouch, right, completely missed that.

OK on keeping the assertion, but I guess just one assertion at the end (the removed one) will be enough?

Reply:
like 537d394a for example

Reply:
I would keep the one in `code_size` loop, too. If something goes wrong, it's easier to debug when it's detected early.

Reply:
okay, then let's keep all, one for each cast to uint16 (one was missing originally)